src/main/groovy/com/readytalk/gradle/plugins/ci/CiInfoPlugin.groovy

```diff
@@ -0,0 +1,174 @@
+package com.readytalk.gradle.plugins.ci
+
+import com.readytalk.gradle.util.PluginUtils
+import com.readytalk.gradle.util.StringUtils
+import nebula.plugin.info.InfoBrokerPlugin
+import nebula.plugin.info.basic.BasicInfoPlugin
+import nebula.plugin.info.java.InfoJavaPlugin
+import nebula.plugin.info.reporting.InfoJarManifestPlugin
+import nebula.plugin.info.reporting.InfoJarPropertiesFilePlugin
+import nebula.plugin.info.reporting.InfoPropertiesFilePlugin
+import nebula.plugin.info.scm.ScmInfoPlugin
+import org.eclipse.jgit.lib.Repository
+import org.eclipse.jgit.lib.RepositoryBuilder
+import org.gradle.api.Plugin
+import org.gradle.api.Project
+import org.gradle.api.publish.ivy.tasks.GenerateIvyDescriptor
+
+class CiInfoPlugin implements Plugin<Project>, PluginUtils {
+  static final String EXTENSION_NAME = 'buildEnv'
+  Project project
+
+  //TODO: Consider using GrGit instead of JGit
+  Repository gitRepo
+  CiInfoExtension extension
+
+  void apply(final Project project) {
+    this.project = project
+    this.extension = new CiInfoExtension()
+    this.gitRepo = new RepositoryBuilder().findGitDir(project.rootDir).build()
+
+    applyNebulaInfoPlugins()
+    addCiInfoExtension()
+    setDefaults()
+    populateCiInfo(System.getenv())
+    mapInfoBrokerFields()
+    configureIvyPublish()
+    configureArtifactory()
+  }
+
+  private void applyNebulaInfoPlugins() {
+    project.plugins.with {
+      //Enables a multitude of reporting plugins, safe to always enable
+      //Applied piecemeal because the CI collector plugin is too simple
+      apply(InfoBrokerPlugin)
+      apply(BasicInfoPlugin)
+      apply(ScmInfoPlugin)
+      apply(InfoJavaPlugin)
+      apply(InfoPropertiesFilePlugin)
+      apply(InfoJarPropertiesFilePlugin)
+      apply(InfoJarManifestPlugin)
+    }
+  }
+
+  private void addCiInfoExtension() {
+    project.extensions.add(EXTENSION_NAME, this.extension)
+  }
+
+  private void setDefaults() {
+    extension.buildNumber = project.plugins.findPlugin('info-ci')?.extension?.buildNumber ?: 'local'
+    extension.branch = gitRepo.branch
+    extension.masterBranch = isMaster(extension.branch)
+    extension.releaseBranch = isReleaseBranch(extension.branch)
+    extension.release = isReleaseTag(extension.branch)
+  }
+
+  private boolean refResolvesToMaster() {
+    gitRepo.getRef('refs/heads/master')?.objectId?.name == gitRepo.resolve('HEAD')?.name
```

Comment:
What do you think would be the best course of action to correct that? I am thinking of removing the safe navigation operators and catching a `NullPointerException` that logs an error and rethrows an appropriate Gradle exception to stop the build.